diff -ruPN linux/arch/arm/boot/dts/rockchip/rk322x-box-v884k.dts linux-new/arch/arm/boot/dts/rockchip/rk322x-box-v884k.dts
--- linux/arch/arm/boot/dts/rockchip/rk322x-box-v884k.dts	1970-01-01 01:00:00.000000000 +0100
+++ linux-new/arch/arm/boot/dts/rockchip/rk322x-box-v884k.dts	2023-12-02 07:40:25.720338993 +0100
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+#include "rk322x-box.dtsi"
+#include "rk322x-opp.dtsi"
+#include "rk322x-legacy.dtsi"
+
+/ {
+	model = "RK3228 V884K";
+	compatible = "rockchip,rk322x-box-v884k", "rockchip,rk3229";
+
+	leds {
+		compatible = "gpio-leds";
+
+		blue_led {
+			gpios = <&gpio3 RK_PC5 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+};
+
+&cpu0_opp_table {
+
+	opp-1104000000 {
+		status = "okay";
+	};
+
+	opp-1200000000 {
+		status = "okay";
+	};
+
+};
+
+&dmc {
+	status = "okay";
+};
+
+&dmc_opp_table {
+	opp-660000000 {
+		status = "okay";
+	};
+};
